### PR TITLE
Remove passwd_changes and user_data from event callbacks

### PR DESCRIPTION
**Pull request by theopolis**
This is a WIP set of changes that will eventually allow subscribers bound to configuration data to update state when the config state updates.

### CODE REVIEW

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by theopolis** [comment]
Need to remove the `passwd_changes` tables/queries from packs.

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

File: include/osquery/events.h [hunk 1]
```diff
@@ -745,21 +847,23 @@ class EventSubscriber : public EventSubscriberPlugin {
...
-    // Down-cast the pointer to the member function.
-    auto base_entry =
-        reinterpret_cast<Status (T::*)(const EventContextRef&, void const*)>(
-            entry);
     // Create a callable through the member function using the instance of the
     // EventSubscriber and a single parameter placeholder (the EventContext).
     auto cb = std::bind(base_entry, sub, _1, _2);
     // Add a subscription using the callable and SubscriptionContext.
-    EventFactory::addSubscription(getType(), sub->getName(), sc, cb, user_data);
+    EventFactory::addSubscription(getType(), this->getName(), sc, cb);
```
Comment (theopolis) [inline include/osquery/events.h]: Although `getName` is a virtual, this should read `sub->getName()`.

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Pull request merged**